### PR TITLE
Rewrite source code to use typescript

### PR DESCRIPTION
**Pull request by jonnyelliot**
Update info and license file
Add script for performance tests
Add Github workflows

### CODE REVIEW

**Comment by jonnyelliot** [comment]
Both sonar and size/size fail due to both not having run on the master branch. The PR is building despite the failing checks.

File: LICENSE [hunk 1]
```diff
@@ -0,0 +1,21 @@
+MIT License
+
+Copyright (c) 2021 Committed Software
```
Comment (stuarthendren) [inline LICENSE:3]: This need the original copy right notice

I suggest you add to the bottom:
```
This incorporates work covered by the following copyright and permission notice:  

HIS FULL LICIENCE
```

File: package.json [hunk 2]
```diff
@@ -1,29 +1,39 @@
 {
-  "name": "trie-search",
-  "author": "Joshua Jung <joshua.p.jung@gmail.com>",
+  "name": "@commitd/trie-search",
```
Comment (stuarthendren) [inline package.json]: our npm package is `@committed`

```suggestion
  "name": "@committed//trie-search",
```

File: package.json [hunk 3]
```diff
@@ -1,29 +1,39 @@
...
+  "author": "Jonathan Elliot <jon@committed.io>",
```
Comment (stuarthendren) [inline package.json]: Hmmm - maybe leave out for now?

File: src/trie-search.ts [hunk 4]
```diff
@@ -0,0 +1,483 @@
+import get from "lodash.get"
+// @ts-ignore
+import HashArray from "hasharray"
+
+var MAX_CACHE_SIZE = 64
```
Comment (stuarthendren) [inline src/trie-search.ts]:
```suggestion
const MAX_CACHE_SIZE = 64
```

File: src/trie-search.ts [hunk 5]
```diff
@@ -0,0 +1,483 @@
...
+
+var IS_WHITESPACE = /^[\s]*$/
```
Comment (stuarthendren) [inline src/trie-search.ts]:
```suggestion
const IS_WHITESPACE = /^[\s]*$/
```

File: src/trie-search.ts [hunk 6]
```diff
@@ -0,0 +1,483 @@
...
+
+var DEFAULT_INTERNATIONALIZE_EXPAND_REGEXES = [
```
Comment (stuarthendren) [inline src/trie-search.ts]:
```suggestion
const DEFAULT_INTERNATIONALIZE_EXPAND_REGEXES = [
```

File: src/trie-search.ts [hunk 7]
```diff
@@ -0,0 +1,483 @@
...
+
+  public reset() {
+    this.root = {}
+    this.size = 0
+  }
+
+  public clearCache() {
+    // if (this.getCache && !this.getCache._list.length) {
+    //   return;
+    // }
```
Comment (stuarthendren) [inline src/trie-search.ts]: ??

**Review by stuarthendren — CHANGES_REQUESTED** [review]
Only the LICENSE and package.json name comments are important.

File: package.json [hunk 8]
```diff
@@ -1,6 +1,6 @@
 {
-  "name": "@commitd/trie-search",
-  "author": "Jonathan Elliot <jon@committed.io>",
+  "name": "@committed/trie-search",
+  "author": "Joshua Jung <joshua.p.jung@gmail.com",
```
Comment (stuarthendren) [inline package.json]:
```suggestion
```

**Review by stuarthendren — APPROVED** [review]

**Pull request merged**